collate.go

```diff
@@ -56,8 +57,15 @@ func (c *JSONCollator) Collate(key1, key2 interface{}) int {
 	case kString:
 		return c.compareStrings(key1.(string), key2.(string))
 	case kArray:
-		array1 := key1.([]interface{})
-		array2 := key2.([]interface{})
+		// Handle the case where a walrus bucket is returning a []float64
+		array1, ok1 := key1.([]interface{})
```

Comment:
It would be cleaner to do this check at the top of toSliceOfInterface() - what do you think?

Reply:
Refactored to toSliceOfInterface